src/sds.c

```diff
@@ -281,6 +285,17 @@ sds sdsMakeRoomFor(sds s, size_t addlen) {
     return s;
 }
 
+/* Enlarge the free space at the end of the sds string more than needed,
+ * pre-allocate at least `sdslen(s) + addlen * 2` bytes of memory. */
+sds sdsMakeRoomFor(sds s, size_t addlen) {
+    return _sdsMakeRoomFor(s, addlen, 0);
+}
+
+/* Unlike sdsMakeRoomFor(), this is just prealloc `addlen` bytes of memory. */
+sds sdsMakeRoomForExact(sds s, size_t addlen) {
```

Comment:
i'm having second thoughts about the "exact" term i specified.
it was good to emphasize the design, but i'd like to suggest another term.
```suggestion
/* Unlike sdsMakeRoomFor(), this is just add the grow to the necessary size. */
sds sdsMakeRoomForNonGreedy(sds s, size_t addlen) {
```